### DISCUSSION TITLE
width and height issue

### DISCUSSION THREAD

**Discussion by Usman-Amjad** [TomSchimansky/CustomTkinter#1592] [Q&A]
There is an issue with width and height.  Whenever I tried to add width and height in .place().  It throws an error that write them in argument but it was not in the previous versions and I got trouble with my software kindly fix it.

**Comment by MaxNiark** [comment]
hello, about which one widget?

.configure will work better ?

VAR_WIDGET.configure(width=X,height=X)

**Reply by Usman-Amjad** [reply]
Yeah Thanks I just didn't want to use another line that's why I asked because in the previous versions you can do that any way you want
Thanks for your answer

**Reply by Usman-Amjad** [reply]
It's not working in the below p_Frame widget
import customtkinter as ctk
from tkinter import *
from tkinter import ttk


class testing:
    def __init__(self, root):
        self.root = root
        self.root.geometry("1425x700+50+20")

        style = ttk.Style(self.root)

        # ====== Product Details ======
        self.p_Frame = ctk.CTkFrame(self.root)
        self.p_Frame.place(x=500, y=185)
        self.p_Frame.configure(width=400, height=485)

        style.configure("Treeview", background="#333333", foreground="white", fieldbackground="#333333", rowheight=30,
                        font=("Arial", 18))
        style.map("Treeview", background=[("selected", "#0078D7")])
        style.configure("Treeview.Heading", font=('Constantia', 18))
        style.layout("Treeview", [('Treeview.treearea', {'sticky': 'nswe'})])  # Remove the borders

        self.product_table = ttk.Treeview(self.p_Frame, style="Treeview", columns=(
            "pid", "category", "name", "scheme", "price", "sellingPrice", "qty", "totalPrice", "status", "location"))
        for column in self.product_table["columns"]:
            self.product_table.column(column, anchor=CENTER)

        scrolly = ctk.CTkScrollbar(self.p_Frame, orientation=VERTICAL, command=self.product_table.yview)
        scrolly.pack(side=RIGHT, fill=Y)
        self.product_table.configure(yscrollcommand=scrolly.set)

        scrollx = ctk.CTkScrollbar(self.p_Frame, orientation=HORIZONTAL, command=self.product_table.xview)
        scrollx.pack(side=BOTTOM, fill=X)
        self.product_table.configure(xscrollcommand=scrollx.set)

        self.product_table.heading("pid", text="Id")
        self.product_table.heading("category", text="Category")
        self.product_table.heading("name", text="Name")
        self.product_table.heading("scheme", text="Scheme")
        self.product_table.heading("price", text="Price")
        self.product_table.heading("sellingPrice", text="Selling Price")
        self.product_table.heading("qty", text="Qty")
        self.product_table.heading("totalPrice", text="Total Price")
        self.product_table.heading("status", text="Status")
        self.product_table.heading("location", text="Location")

        self.product_table["show"] = "headings"

        self.product_table.column("pid", width=80, minwidth=80)
        self.product_table.column("category", width=140, minwidth=140)
        self.product_table.column("name", width=140, minwidth=140)
        self.product_table.column("scheme", width=100, minwidth=100)
        self.product_table.column("price", width=100, minwidth=100)
        self.product_table.column("sellingPrice", width=140, minwidth=140)
        self.product_table.column("qty", width=100, minwidth=100)
        self.product_table.column("totalPrice", width=140, minwidth=140)
        self.product_table.column("status", width=100, minwidth=100)
        self.product_table.column("location", width=120, minwidth=120)

        self.product_table.pack(fill=BOTH, expand=1)


if __name__ == "__main__":
    root = ctk.CTk()
    obj = testing(root)
    root.mainloop()

**Reply by zSynctic** [reply]
In your original question you said adding height and width into .place() function but that's wrong you need to put the coordinates in x and y,
in your last message can you explain what is the issue?

**Reply by Usman-Amjad** [reply]
In the above code, I am trying to configure width and height in self.p_Frame. But it's not doing anything in my gui

**Comment by MaxNiark** [comment]
Ok some issue here : 

=> super().__init__() is missing
=> i don't know if root = ctk.CTk() is important in your code, if it's not take my code. 

i did deleted the style cuz i would like to get natural widget from CTK, u can add it 
have fun

```
import customtkinter
from customtkinter import *
from tkinter import *
from tkinter import ttk


class testing(CTk):
    def __init__(self):
        super().__init__()


        self.p_Frame = CTkFrame(self, fg_color="RED")
        self.p_Frame.configure(width=400, height=86)
        self.p_Frame.place(x=1,y=1)
        self.product_table = ttk.Treeview(self.p_Frame, style="Treeview", columns=(
            "pid", "category", "name", "scheme", "price", "sellingPrice", "qty", "totalPrice", "status", "location"))
        for column in self.product_table["columns"]:
            self.product_table.column(column, anchor=CENTER)

        scrolly = CTkScrollbar(self.p_Frame, orientation=VERTICAL, command=self.product_table.yview)
        scrolly.pack(side=RIGHT, fill=Y)
        self.product_table.configure(yscrollcommand=scrolly.set)

        scrollx = CTkScrollbar(self.p_Frame, orientation=HORIZONTAL, command=self.product_table.xview)
        scrollx.pack(side=BOTTOM, fill=X)
        self.product_table.configure(xscrollcommand=scrollx.set)

        self.product_table.heading("pid", text="Id")
        self.product_table.heading("category", text="Category")
        self.product_table.heading("name", text="Name")
        self.product_table.heading("scheme", text="Scheme")
        self.product_table.heading("price", text="Price")
        self.product_table.heading("sellingPrice", text="Selling Price")
        self.product_table.heading("qty", text="Qty")
        self.product_table.heading("totalPrice", text="Total Price")
        self.product_table.heading("status", text="Status")
        self.product_table.heading("location", text="Location")

        self.product_table["show"] = "headings"

        self.product_table.column("pid", width=80, minwidth=80)
        self.product_table.column("category", width=140, minwidth=140)
        self.product_table.column("name", width=140, minwidth=140)
        self.product_table.column("scheme", width=100, minwidth=100)
        self.product_table.column("price", width=100, minwidth=100)
        self.product_table.column("sellingPrice", width=140, minwidth=140)
        self.product_table.column("qty", width=100, minwidth=100)
        self.product_table.column("totalPrice", width=140, minwidth=140)
        self.product_table.column("status", width=100, minwidth=100)
        self.product_table.column("location", width=120, minwidth=120)

        self.product_table.pack(fill=BOTH, expand=1)
name = "main"
if name == "main":

    obj = testing()
    obj.mainloop()

```
